torchdata/nodes/samplers/multi_node_weighted_sampler.py

```diff
@@ -63,21 +65,31 @@ def __init__(
         seed: int = 0,
     ) -> None:
         super().__init__()
+
+        # Initialize basic attributes
         self.source_nodes = source_nodes
         self.weights = weights
         self.stop_criteria = stop_criteria
         self.dataset_names = list(self.source_nodes.keys())
         self._num_yielded = 0
+        self._started = False
         self.seed = seed
 
+        # Setup rank and world size
         if rank is None or world_size is None:
             self.rank, self.world_size = get_rank_and_world_size()
         else:
             self.rank = rank
             self.world_size = world_size
 
+        self._epoch_random_seed = _update_epoch_seed(self.seed * self.world_size + self.rank)
```

Comment:
```suggestion
        self._epoch = 0
```

Would this ^ still work?